src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -637,13 +667,50 @@ public abstract class GenerateProtoTask extends DefaultTask {
 
   protected String computeExecutablePath(ExecutableLocator locator) {
     if (locator.path != null) {
-      return locator.path
+      return locator.path.endsWith(JAR_SUFFIX) ? createJarTrampolineScript(locator.path) : locator.path
     }
     File file = locatorToAlternativePathsMapping.getting(locator.name).get().singleFile
+    if (file.name.endsWith(JAR_SUFFIX)) {
+      return createJarTrampolineScript(file.getAbsolutePath())
+    }
+
     if (!file.canExecute() && !file.setExecutable(true)) {
       throw new GradleException("Cannot set ${file} as executable")
     }
     logger.info("Resolved artifact: ${file}")
     return file.path
   }
+
+  /**
+   * protoc only supports plugins that are a single self contained executable file. For .jar files create a trampoline
+   * script to execute the jar file. Assume the jar is a "fat jar" or "uber jar" and don't attempt any artifact
+   * resolution.
+   * @param jarAbsolutePath Absolute path to the .jar file.
+   * @return The absolute path to the trampoline executable script.
+   */
+  private String createJarTrampolineScript(String jarAbsolutePath) {
+    assert jarAbsolutePath.endsWith(JAR_SUFFIX)
+    boolean isWindows = isWindows()
+    File scriptExecutableFile = new File("${project.buildDir}/scripts/" +
+            jarAbsolutePath[0..(jarAbsolutePath.length() - JAR_SUFFIX.length())] + "-trampoline." +
+            (isWindows ? "bat" : "sh"))
+    try {
+      if (createNewScriptFile(scriptExecutableFile)) {
+        String javaExe = computeJavaExePath(isWindows)
+        // Rewrite the trampoline file unconditionally (even if it already exists) in case the dependency or versioning
+        // changes we don't need to detect the delta (and the file content is cheap to re-generate).
+        new FileOutputStream(scriptExecutableFile).withCloseable { execOutputStream ->
```

Comment:
nit: `scriptExecutableFile.setBytes(...)` would also work and be a bit more natural.